packages/jss/src/RuleList.js

```diff
@@ -114,7 +114,10 @@ export default class RuleList {
   remove(rule) {
     this.unregister(rule)
     delete this.raw[rule.key]
-    this.index.splice(this.index.indexOf(rule), 1)
```

Comment:
why is by reference not working?

Reply:
In fact, it should work with `indexOf`, while only adding the `-1` check, right? If help is needed, I can try myself. What else is needed to merge the PR? Some tests?

Reply:
I lack the greater context to effectively debug the reason why is 'find by reference' not working. 

> In fact, it should work with indexOf, while only adding the -1 check, right? 

If we checked only for -1 it wouldn't be removing old styles

Reply:
> why is by reference not working?

I am not quite sure, but I think it has to do with the following:

![{7233EC94-9A76-4DF6-8DDE-CA49C6CD6B0B}](https://github.com/user-attachments/assets/df2d2386-828b-40ae-98f1-553652cc1aa8)

When some dynamic rules are updated because the data might change, those new updated rules are not stored in the [useStyles](https://github.com/cssinjs/jss/blob/master/packages/react-jss/src/createUseStyles.js) hook. So, finally when `removeDynamicRules` happens, we remove the old rule references, instead of the new ones. I think this fix handles that correctly, but some new test cases should be added. Would be very nice if a new version would be released soon, since JSS is still widely used.